File: hackerrank/easy/0039-repeated-string/python/solution.py
## solution.py

def repeated_string(s: str, n: int) -> int:
    full_repeat = n // len(s)
    leftover = n % len(s)
    a_full_count = s.count('a')
    a_leftover_count = s[:leftover].count('a')
    return full_repeat * a_full_count + a_leftover_count


# --- demo runs ---
if __name__ == "__main__":
    print(repeated_string("aba", 10))           # 7
    print(repeated_string("a", 1000000000000))  # 1000000000000

File: hackerrank/easy/0039-repeated-string/swift/solution.swift
// solution.swift

import Foundation

func repeatedString(s: String, n: Int) -> Int {
    let fullRepeat = n / s.count
    let leftover = n % s.count
    let aFullCount = s.filter { $0 == "a" }.count
    let aLeftoverCount = s.prefix(leftover).filter { $0 == "a" }.count
    return fullRepeat * aFullCount + aLeftoverCount
}

// --- demo runs ---
print(repeatedString(s: "aba", n: 10))          // 7
print(repeatedString(s: "a", n: 1000000000000)) // 1000000000000